python/ccf/clients.py

```diff
@@ -572,28 +563,32 @@ def __init__(
         host: str,
         port: int,
         ca: str,
-        cert: Optional[str] = None,
-        key: Optional[str] = None,
+        session_auth: Optional[Identity] = None,
+        signing_auth: Optional[Identity] = None,
         connection_timeout: int = DEFAULT_CONNECTION_TIMEOUT_SEC,
         description: Optional[str] = None,
         ws: bool = False,
-        disable_client_auth: bool = False,
     ):
         self.connection_timeout = connection_timeout
-        self.description = description
         self.name = f"[{host}:{port}]"
+        self.description = description
+        if self.description is not None:
+            if session_auth is not None:
+                self.description += "a"
+            if signing_auth is not None:
+                self.description += "s"
+        else:
+            self.description = self.name
```

Comment:
This produces prefixes like:

```
[0|member0]as
```

So the `a` is just repeating the fact that `member0` is there. I think this should be rolled into the construction of the initial description in `node.py`. If we have different auth and signing identities, they should both be present. Something like:

```
[0|member1] # I'm talking to node 0 as member1
[0|member1s] # I'm talking to node 0 as member1, and signing everything
[0|member1|member2s] # I'm talking to node 0 as member1, and signing everything as member2
```

Reply:
Yeah good idea.

Reply:
I'm going with positions instead, the s suffix inside the bracket could be part of the name, I don't think it's a very happy choice.